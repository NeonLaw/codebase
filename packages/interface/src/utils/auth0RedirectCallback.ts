import { navigate } from 'gatsby';

export const onRedirectCallback = (appState) => {
  return navigate(appState?.returnTo || '/');
};
