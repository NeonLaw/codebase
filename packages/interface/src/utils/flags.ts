interface FlagsInterface {
  ACCESSIBLE_BUTTONS: boolean;
}

export const defaultFlags: FlagsInterface = {
  ACCESSIBLE_BUTTONS: true,
};

export const flags = (userFlagsString: string): FlagsInterface => {
  const userFlags = {};

  userFlagsString.split(',').forEach((flag) => {
    if (flag[0] === '-') {
      userFlags[flag.substr(1)] = false;
    } else {
      userFlags[flag] = true;
    }
  });

  return {...defaultFlags, ...userFlags};
};
