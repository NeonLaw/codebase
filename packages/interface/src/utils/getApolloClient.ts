import { ApolloClient, InMemoryCache, createHttpLink } from '@apollo/client';
import fetch from 'isomorphic-fetch';
import { setContext } from '@apollo/client/link/context';

export const getApolloClient = (
  getAccessTokenSilently: (
    options?: any | undefined
  ) => Promise<string>
) => {
  const authLink = setContext(async (_, { headers }) => {
    const token = await getAccessTokenSilently();
    if (!token) {
      return { headers };
    }
    return {
      headers: {
        ...headers,
        authorization: `Bearer ${token}`,
      },
    };
  });

  const httpLink = createHttpLink({
    fetch,
    uri: process.env.GATSBY_API_URL,
    useGETForQueries: false,
  });

  return new ApolloClient({
    cache: new InMemoryCache(),
    link: authLink.concat(httpLink),
  });
};
