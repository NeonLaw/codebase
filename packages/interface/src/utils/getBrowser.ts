export const getBrowser = () => {
  const userAgent =
    typeof window !== undefined ? window.navigator.userAgent : '';
  const browsers = ['Opera', 'Chrome', 'Firefox', 'IE'];
  let browser;

  browsers.forEach((b) => {
    if (userAgent.indexOf(b) !== -1) {
      browser = b;
    }
  });

  return browser;
};
