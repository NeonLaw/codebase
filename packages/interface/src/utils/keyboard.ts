const submit = (e: any, ref: any) => {
  const form = ref.current;
  e.preventDefault();
  if (null !== form) {
    form.dispatchEvent(new Event('submit', { cancelable: true }));
  }
};

export const submitOnShiftEnter = (e: any, ref: any) => {
  if (isShiftEnterPressed(e)) {
    submit(e, ref);
  }
};

export const submitOnMetaEnter = (e: any, ref: any) => {
  if (isCommandEnterPressed(e)) {
    submit(e, ref);
  }
};

export const isShiftEnterPressed = (e) => e.shiftKey && e.key == 'Enter';

export const isCommandEnterPressed = (e) => e.metaKey && e.key == 'Enter';
