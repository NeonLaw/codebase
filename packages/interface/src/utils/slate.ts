import { Node } from 'slate';

export const  convertSlateToPlaintext = (slateJson: any) => {
  return slateJson.map(n => Node.string(n)).join('\n');
};
