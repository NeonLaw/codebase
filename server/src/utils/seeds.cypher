MATCH (a:Questionnaire),(b:Question)
WHERE a.name = 'Community Property' AND b.prompt = 'Character Evidence'
CREATE (a)-[r:NEXT_QUESTION { rule: "*" }]->(b)
RETURN type(r);

MATCH (a:Question),(b:Question)
WHERE a.prompt = 'Character Evidence' AND b.prompt = 'Non-Hearsay'
CREATE (a)-[r:NEXT_QUESTION { rule: "*" }]->(b)
RETURN type(r);

MATCH (a:Question),(b:Question)
WHERE a.prompt = 'Non-Hearsay' AND b.prompt = 'Hearsay Exception Unavailable Witness'
CREATE (a)-[r:NEXT_QUESTION { rule: "*" }]->(b)
RETURN type(r);
