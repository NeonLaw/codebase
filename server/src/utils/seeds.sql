DO $$
DECLARE nevada_c_corp_matter_template_id UUID;
DECLARE primary_contact_id UUID;
BEGIN

INSERT INTO people (email, role) VALUES ('portal@sink.sendgrid.com', 'portal') ON CONFLICT DO NOTHING RETURNING (id) INTO primary_contact_id;
INSERT INTO people (email, role) VALUES ('admin@sink.sendgrid.com', 'admin') ON CONFLICT DO NOTHING;

INSERT INTO matter_templates (name, javascript_module, category) VALUES ('California Bar Prep', '@neonlaw/matter-templates/california/bar-prep', 'bar-prep') ON CONFLICT DO NOTHING;
INSERT INTO matter_templates (name, javascript_module, category) VALUES ('Nevada C Corp', '@neonlaw/matter-templates/nevada/c-corp', 'business') ON CONFLICT DO NOTHING RETURNING (id) INTO nevada_c_corp_matter_template_id;
INSERT INTO matter_templates (name, javascript_module, category) VALUES ('Nevada LLC', '@neonlaw/matter-templates/nevada/llc', 'business') ON CONFLICT DO NOTHING;

INSERT INTO questionnaires (name, matter_template_id) VALUES ('Community Property', (SELECT id FROM matter_template WHERE name = 'California Bar Prep' LIMIT 1)) ON CONFLICT DO NOTHING;

INSERT INTO questions (prompt, question_type, help_text)
VALUES ('Character Evidence', 'text', '[{ "type": "paragraph", "children": [{ "text": "help" }]}]')
ON CONFLICT DO NOTHING;

INSERT INTO questions (prompt, question_type, help_text)
VALUES ('Non-Hearsay', 'text', '[{ "type": "paragraph", "children": [{ "text": "help" }]}]')
ON CONFLICT DO NOTHING;

INSERT INTO questions (prompt, question_type, help_text)
VALUES ('Hearsay Exception Unavailable Witness', 'text', '[{ "type": "paragraph", "children": [{ "text": "help" }]}]')
ON CONFLICT DO NOTHING;

INSERT INTO matters (name, primary_contact_id, matter_template_id) VALUES ('Nevada LLC Demo', primary_contact_id, nevada_c_corp_matter_template_id);
END $$;
