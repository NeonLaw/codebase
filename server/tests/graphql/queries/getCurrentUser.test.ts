import * as faker from 'faker';
import { describe, expect, it } from '@jest/globals';
import { runGraphQLQuery } from '../../utils/graphqlHelpers';

describe('the getCurrentUser query', () => {
  describe('an anonymous user', () => {
    it('returns a null current user object', async () => {
      await runGraphQLQuery({
        async databaseSetup() { return; },
        async databaseTeardown() { return; },
        expectations(response) {
          expect(response.errors).toBeTruthy();
          expect(response.data.getCurrentUser).toBeNull();
          expect(response.errors[0].message).toEqual(
            'permission denied for table person'
          );
        },
        query: '{getCurrentUser { id }}',
        variables: {},
      });
    });
  });

  describe('a portal user', () => {
    it('returns the current user object', async () => {
      const personId = faker.datatype.uuid();

      await runGraphQLQuery({
        async databaseSetup(client) {
          await client.query(
            'INSERT INTO person (id, email, role, sub) ' +
            'VALUES ($1, $2, $3, $4) RETURNING id;',
            [personId, 'test@example.com', 'portal', 'test-portal']
          );
        },
        async databaseTeardown(client) {
          await client.query(
            'DELETE FROM person WHERE id = $1',
            [personId]
          );
        },
        expectations(response) {
          expect(response.errors).toBeFalsy();
          expect(response.data.getCurrentUser.id).toEqual(personId);
        },
        personId,
        query: '{getCurrentUser { id }}',
        role: 'portal',
        variables: {},
      });
    });
  });

  describe('a lawyer user', () => {
    it('returns the current user object', async () => {
      const personId = faker.datatype.uuid();

      await runGraphQLQuery({
        async databaseSetup(client) {
          await client.query(
            'INSERT INTO person (id, email, role, sub) ' +
            'VALUES ($1, $2, $3, $4) RETURNING id;',
            [personId, 'test@example.com', 'lawyer', 'test-lawyer']
          );
        },
        async databaseTeardown(client) {
          await client.query(
            'DELETE FROM person WHERE id = $1',
            [personId]
          );
        },
        expectations(response) {
          expect(response.errors).toBeFalsy();
          expect(response.data.getCurrentUser.id).toEqual(personId);
        },
        personId,
        query: '{getCurrentUser { id }}',
        role: 'lawyer',
        variables: {},
      });
    });
  });

  describe('a admin user', () => {
    it('returns the current user object', async () => {
      const personId = faker.datatype.uuid();

      await runGraphQLQuery({
        async databaseSetup(client) {
          await client.query(
            'INSERT INTO person (id, email, role, sub) ' +
            'VALUES ($1, $2, $3, $4) RETURNING id;',
            [personId, 'test@example.com', 'admin', 'test-admin']
          );
        },
        async databaseTeardown(client) {
          await client.query(
            'DELETE FROM person WHERE id = $1',
            [personId]
          );
        },
        expectations(response) {
          expect(response.errors).toBeFalsy();
          expect(response.data.getCurrentUser.id).toEqual(personId);
        },
        personId,
        query: '{getCurrentUser { id }}',
        role: 'admin',
        variables: {},
      });
    });
  });
});
