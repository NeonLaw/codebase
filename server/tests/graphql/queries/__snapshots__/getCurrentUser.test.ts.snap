// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the getCurrentUser query a admin user returns the current user object 1`] = `
Object {
  "data": Object {
    "getCurrentUser": Object {
      "id": "[id-1]",
    },
  },
}
`;

exports[`the getCurrentUser query a lawyer user returns the current user object 1`] = `
Object {
  "data": Object {
    "getCurrentUser": Object {
      "id": "[id-1]",
    },
  },
}
`;

exports[`the getCurrentUser query a portal user returns the current user object 1`] = `
Object {
  "data": Object {
    "getCurrentUser": Object {
      "id": "[id-1]",
    },
  },
}
`;

exports[`the getCurrentUser query an anonymous user returns a null current user object 1`] = `
Object {
  "data": Object {
    "getCurrentUser": null,
  },
  "errors": Array [
    Object {
      "locations": Array [
        Object {
          "column": 2,
          "line": 1,
        },
      ],
      "message": "permission denied for table person",
      "path": Array [
        "getCurrentUser",
      ],
    },
  ],
}
`;
